new text detect models

dmMaze opened the pull request:
I trained text detection model on Manga109, American comics, and synthetic data, the new model should be able to achieve better results on the English comics, and keep comparable performance on manga.(at least the result looks good on Manga109 & synthetic data) 

Please download the [model](https://drive.google.com/file/d/1Rj-ufYATN9YB8PqiGOj7rcFnvTwjIoHE/view?usp=sharing) and run translate_demo.py with the option --use-ctd.

And implementation of text polygons and mask post-processing based on the new models 

- Angled text lines post-processing involving splitting, merging, sorting, font size and angle estimation, etc, is based on the TextBlock class, the idea is corresponding to the method adopted by origin repo based on Quadrilateral.

- Text mask refinement can be summarized as:
  1. Extract connected components from a list of different representations of the original image
  2. Merge connected components that are close to the predicted mask. (their distance is measured by pixel-wise xor)
Please refer to textmask.py for more details.

zyddnys commented:
Fantastic work! It did outperform in some English comic cases.
However it spits out errors for some images, I have linked them below.
https://cotrans-test-bucket.s3.us-west-1.amazonaws.com/paper1.png
https://cotrans-test-bucket.s3.us-west-1.amazonaws.com/20210420104233.jpg
https://cotrans-test-bucket.s3.us-west-1.amazonaws.com/20210831111741.jpg

dmMaze commented:
> Fantastic work! It did outperform in some English comic cases. However it spits out errors for some images, I have linked them below. https://cotrans-test-bucket.s3.us-west-1.amazonaws.com/paper1.png https://cotrans-test-bucket.s3.us-west-1.amazonaws.com/20210420104233.jpg https://cotrans-test-bucket.s3.us-west-1.amazonaws.com/20210831111741.jpg

Fixed.

I enlarged English text polygons (1/3 font size) for ocr because their detected boxes are  a bit smaller than gt.

There's a potential bug in the translator: some text is not translated,  using '\n' to split translated text would end up incorrect alignment with source text blocks. This can be solved by inserting breakers like '\n###\n' and removing them afterward, maybe there's a better way to do it but idk. I modified for google translator only since I don't have APIs of other translators.
